src/test/java/com/myhome/services/unit/UserSDJpaServiceTest.java

```diff
@@ -0,0 +1,204 @@
+package com.myhome.services.unit;
+
+import com.myhome.controllers.dto.UserDto;
+import com.myhome.controllers.dto.mapper.UserMapper;
+import com.myhome.domain.Community;
+import com.myhome.domain.CommunityAdmin;
+import com.myhome.domain.User;
+import com.myhome.repositories.UserRepository;
+import com.myhome.services.CommunityService;
+import com.myhome.services.springdatajpa.UserSDJpaService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+
+class UserSDJpaServiceTest {
+
+  private String userId = "test-user-id";
+  private String userName = "test-user-id";
+  private String userEmail = "test-user-id";
+  private String userPassword = "test-user-id";
+
+  @Mock
+  private UserRepository userRepository;
+  @Mock
+  private UserMapper userMapper;
+  @Mock
+  private PasswordEncoder passwordEncoder;
+  @Mock
+  private CommunityService communityService;
+  @InjectMocks
+  private UserSDJpaService userService;
+
+  @BeforeEach
+  void setUp() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  void createUserSuccess() {
+    // given
+    UserDto request = getDefaultUserDtoRequest();
+    User resultUser = new User(request.getName(), request.getUserId(), request.getEmail(), request.getEncryptedPassword());
+    UserDto response = new UserDto(
+        resultUser.getId(),
+        resultUser.getUserId(),
+        resultUser.getName(),
+        resultUser.getEmail(),
+        null,
+        resultUser.getEncryptedPassword(),
+        new HashSet<>()
+    );
+
+    given(userRepository.findByEmail(request.getEmail()))
+        .willReturn(null);
+    given(passwordEncoder.encode(request.getPassword()))
+        .willReturn(request.getPassword());
+    given(userMapper.userDtoToUser(request))
+        .willReturn(resultUser);
+    given(userRepository.save(resultUser))
+        .willReturn(resultUser);
+    given(userMapper.userToUserDto(resultUser))
+        .willReturn(response);
+
+    // when
+    Optional<UserDto> createdUserDtoOptional = userService.createUser(request);
+
+    // then
+    UserDto createdUserDto = createdUserDtoOptional.get();
```

Comment:
Let's switch this two lines of code since firstly it should be assertion if optional is present (if not the test will fail) and nextly we can `.get()` the value of it (because if the assertion was correct, then optional is not empty).

This needs to be applicable in other test methods.